include/boost/log/detail/config.hpp

```diff
@@ -98,10 +98,12 @@
 #   define BOOST_LOG_BROKEN_CONSTANT_EXPRESSIONS
 #endif
 
-#if defined(BOOST_NO_CXX11_HDR_CODECVT)
+#if (defined(BOOST_NO_CXX11_HDR_CODECVT) && (BOOST_CXX_VERSION < 201700)) || (defined(_MSVC_STL_VERSION) && (_MSVC_STL_VERSION < 142))
```

Comment:
This didn't work when I tried this change in my local Boost install (I'm pinned against Boost 1.72 for now). I had to change this to:
#if (defined(BOOST_NO_CXX11_HDR_CODECVT) && (_MSVC_LANG < 201703)) || (defined(_MSVC_STL_VERSION) && (_MSVC_STL_VERSION < 142))

In order for BOOST_LOG_NO_CXX11_CODECVT_FACETS to not be defined.
The correct version number of _MSVC_LANG to check against for C++17 is 201703, not 201700, and in my case, BOOST_CXX_VERSION was not defined, so I changed it to look at _MSVC_LANG, instead.

Reply:
I've updated the version check, thanks.